sootup.callgraph/src/main/java/sootup/callgraph/AbstractCallGraphAlgorithm.java

```diff
@@ -447,11 +447,7 @@ public CallGraph addClass(@Nonnull CallGraph oldCallGraph, @Nonnull JavaClassTyp
    */
   public MethodSignature findMainMethod() {
     Set<SootClass> classes = new HashSet<>(); /* Set to track the classes to check */
-    for (SootClass aClass : view.getClasses()) {
-      if (!aClass.isLibraryClass()) {
-        classes.add(aClass);
-      }
-    }
+    classes.addAll(view.getClasses().filter(aClass -> !aClass.isLibraryClass()).collect(Collectors.toSet()));
 
     Collection<SootMethod> mainMethods = new HashSet<>(); /* Set to store the methods */
     for (SootClass aClass : classes) {
```

Comment:
next iteration: lets not collect and store them classes in a HashSet - use the stream directly in the loop i.e. we can transform the loop to make use of streams as well :)

Reply:
I tried to get rid of the two for loops. I hope this was correct. (First time contributing to open source, so no idea if what I'm doing is correct)

Reply:
That looks good! We just don't need to create the Pair as we can get the type of the declaring class via methodSignature.getDeclClassType(). Don't be afraid of doing something wrong - we have version control to tackle that ;) So everything is a opportunity to learn and as long as the communication is good a lot of things are possible :)

Reply:
which is not even necessary as we can work on the MethodSubSignature - JavaIdentifierFactory.getInstance().isMainSubSignature( method.getSignature().getSubSignature() ) will do the job of everything that is inside the equals part of the condition